include/nuttx/lin.h

```diff
@@ -44,45 +44,43 @@
 #define LIN_RTR_FLAG              CAN_RTR_FLAG  /* Describe the direction of sending and receiving */
 #define LIN_ERR_FLAG              CAN_ERR_FLAG  /* The flag indicate  this is LIN err_frame */
 
-/* Bit flags on can_frame.types
- *
- * Use can_frame.types to identify other special frame.
- *
- * LIN_TCF_FLAG : TxConfirmation message frame. Lower_half use this flag to
- *                confirm frame-transmit
- *
- * LIN_EVT_FLAG : Event message frame. Lower_half use this flag to report
- *                state switch event
- *
- * LIN_CACHE_RESP_FLAG: When slave response to master, slave node should send
- *                      frame immediately which already be cached in last
- *                      transmission in case of response interval over time.
- *
- * LIN_CHKSUM_EXT_FLAG: LIN checksum have two types, default type will be
- *                      classic checksum
- *
- * LIN_SINGLE_RESP_FLAG: Cache LIN frame only work once, then will be cleared
+/* When slave response to master, slave node should send  frame immediately
+ * which already be cached in last transmission in case of response interval
+ * over time
  */
 
-#define LIN_TCF_FLAG              CAN_TCF_FLAG
-#define LIN_EVT_FLAG              CAN_EVT_FLAG
-#define LIN_CACHE_RESP_FLAG       (1 << (CANFD_FLAGS_BITS + 2))
-#define LIN_CHKSUM_EXT_FLAG       (1 << (CANFD_FLAGS_BITS + 3))
-#define LIN_SINGLE_RESP_FLAG      (1 << (CANFD_FLAGS_BITS + 4))
+#define LIN_CACHE_RESPONSE        (1 << (LIN_ID_BITS))
+
+/* LIN checksum have two types, default type will be classic checksum */
+
+#define LIN_CHECKSUM_EXTENDED     (1 << (LIN_ID_BITS + 1))
+
+/* Cache LIN frame only work once. then will be clear */
+
+#define LIN_SINGLE_RESPONSE       (1 << (LIN_ID_BITS + 2))
+
+/* Lower_half use this flags to report state switch event */
+
+#define LIN_EVT_FLAG              (1 << (LIN_ID_BITS + 3))
```

Comment:
why not map to CAN_EVT_FLAG

Reply:
The location where CAN_EVT_FLAG is placed occupies the location of the expanded can frame's ID and needs to be removed later. Now consider that lin has more available special locations, so it is decoupled from the can flag.In this way, Lin will not change with the changes of can.

Reply:
CAN_EVT_LANG is located at position 28 of can_id, which should be the can_id value. This is an incorrect definition and we will remove it.
Considering that the ID lengths used for CAN and LIN are different, defining special frames for CAN and LIN separately will make the code more concise.